private-templates-service/client/src/components/Common/VersionModal/VersionModal.tsx

```diff
@@ -0,0 +1,160 @@
+
+// Libraries
+import { Checkbox } from 'office-ui-fabric-react/lib/Checkbox';
+
+import { Template, PostedTemplate, TemplateInstance } from 'adaptive-templating-service-typescript-node';
+
+// Redux
+import { updateTemplate, deleteTemplateVersion } from '../../../store/currentTemplate/actions';
+
+// Styles
+import {
+  BackDrop,
+  Modal,
+  Header,
+  Description,
+  DescriptionAccent,
+  CenterPanelWrapper,
+  BottomRow,
+  ButtonGroup,
+  LightButton,
+  PrimaryStyleButton, 
+  Card, 
+  CardHeaderRow,
+  CardHeaderText, 
+  CardBody,
+  VersionWrapper,
+  DateWrapper,
+  StatusWrapper,
+  CheckboxWrapper
+} from './styled';
+
+import {
+  VersionCardRow,
+} from '../../Dashboard/PreviewModal/TemplateInfo/VersionCard/styled';
+
+import {
+  StatusIndicator,
+  Status
+} from '../../Dashboard/PreviewModal/TemplateInfo/styled';
+
+import { getDateString } from '../../../utils/versionUtils';
+import ModalHOC from '../../../utils/ModalHOC';
+
+interface Props {
+  template: Template;
+  toggleModal: () => void;
+  updateTemplateState: (templateJSON: string, state: PostedTemplate.StateEnum, version: string) => void;
+  deleteTemplateVersion: (version: string) => void;
+}
+
+const mapDispatchToProps = (dispatch: any) => {
+  return {
+    updateTemplateState: (templateJSON: string, state: PostedTemplate.StateEnum, version: string) => {
+      dispatch(updateTemplate(undefined, version, templateJSON, undefined, undefined, state, undefined));
+    },
+    deleteTemplateVersion: (version: string) => {
+      dispatch(deleteTemplateVersion(version, undefined))
+    }
+  }
+}
+
+interface State {
+  versionList: string[];
+}
+
+class VersionModal extends React.Component<Props, State> {
+  constructor(props: Props){
+    super(props);
+    this.state = { versionList: [] }
+  }
+  
+  delete = () => {
+    for (let instance of this.props.template.instances!) {
+      if (!instance.version || !this.state.versionList.includes(instance.version!)) continue;
+      this.props.deleteTemplateVersion(instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  publish = () => {
+    for (let instance of this.props.template.instances!){
+      if (!this.state.versionList.includes(instance.version!)) continue;
+      const template = JSON.stringify(instance.json);
+      this.props.updateTemplateState(template, PostedTemplate.StateEnum.Live, instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  unpublish = () => {
+    for (let instance of this.props.template.instances!){
+      if (!this.state.versionList.includes(instance.version!)) continue;
+      const template = JSON.stringify(instance.json);
+      this.props.updateTemplateState(template, PostedTemplate.StateEnum.Deprecated, instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  addVersion = (isChecked?: boolean, version?: string) => {
+    if (!isChecked || !version) return;
+    let versionList = this.state.versionList;
+    if (isChecked === true) {
```

Comment:
would'nt that work? if(isChecked)?

Reply:
The parameter checked is from checkbox, but since this value might be undefined, I pass in isChecked as an optional parameter and I only want it to be added into the state's version list if isChecked is defined and true and removed otherwise